### DISCUSSION TITLE
How do I load multiple map models in opentcs? Can they upload separately?

### DISCUSSION THREAD

**Discussion by CSXYA** [openTCS/opentcs#86] [Q&A] [answered]
In one scene, it will have multiple maps, and then AGVs will run around in multiple maps, can I load multiple map models in open tcs? Can open tcs map models be uploaded separately? For example, I upload only one vehicle information, and then pass in the station and other information

**Comment by martingr** [comment]
No, uploading partial plant models into the kernel is not possible. In openTCS, a plant model loaded into the kernel usually contains everything needed for operation. What would be your use case for uploading some of the information seperately?

**Reply by CSXYA** [reply]
I want to connect to the agv at startup, but I may not have the site information yet. What can I do to load the map model at startup?

**Reply by swltr** [reply]
The complete plant model must be uploaded to the kernel before operating the plant. The plant model cannot be extended during operation.

If all you want is to get plant model data from AGVs, you could implement a `PlantModelImporter` for the Model Editor application (see [Developer's Guide](https://www.opentcs.org/docs/5.8/developer/developers-guide/opentcs-developers-guide.html#_how_to_add_importexport_functionality_for_plant_model_data)). This importer could connect to one or more vehicles, get map data from them and create a plant model from the map data, which you then upload to the kernel. But still, this has to happen before, not during, operation.

**Reply by CSXYA** [reply]
Can I create multiple map models in the project?

**Reply by CSXYA** [reply]
What is the trigger mode for kernel to read map model

**Reply by swltr** [reply, marked as answer]
The kernel always works with exactly one plant model. If you have multiple maps, e.g. for different AGVs working in different areas, you need to merge them into a single plant model for openTCS.

Programmatically, [PlantModelService.createPlantModel()](https://www.opentcs.org/docs/5.8/developer/api-base/org/opentcs/components/kernel/services/PlantModelService.html#createPlantModel(org.opentcs.access.to.model.PlantModelCreationTO)) is used to load a plant model into the kernel.

**Reply by CSXYA** [reply]
thank you
